scripts/subscribeEventProof.js

```diff
@@ -8,99 +8,53 @@ let txExecutor;
-                        return resolve();
-                    }
-                    const {eventId, validatorSetId, signatures} = eventProof;
-                    let v = [], r = [], s = []; // signature params
-                    signatures.forEach(signature => {
-                        const hexifySignature = ethers.utils.hexlify(signature);
-                        const sig = ethers.utils.splitSignature(hexifySignature);
-                        v.push(sig.v);
-                        r.push(sig.r);
-                        s.push(sig.s);
-                    });
-                    const cennznetEventProof = {
-                        eventId,
-                        validatorSetId,
-                        v,
-                        r,
-                        s
-                    };
-
-                    logger.info("Sending setValidators tx with the account:", txExecutor.address);
-                    const gasEstimated = await bridge.estimateGas.setValidators(newValidators, cennznetEventProof, {gasLimit: 500000});
-                    logger.info(await bridge.setValidators(newValidators, cennznetEventProof, {gasLimit: 500000}));
-                    const balance = await ethers.provider.getBalance(txExecutor.address);
-                    logger.info(`Balance is: ${balance}`);
-                    const gasPrice = await ethers.provider.getGasPrice();
-                    logger.info(`Gas price: ${gasPrice.toString()}`);
-                    const gasRequired = gasEstimated.mul(gasPrice);
-                    logger.info(`Gas required: ${gasRequired.toString()}`);
-                    if (balance.lt(gasRequired.mul(2))) {
-                        const { statusCode, data } = await curly.post(`https://hooks.slack.com/services/${process.env.SECRET}`, {
-                            postFields: JSON.stringify({
-                                "text": ` 🚨 To keep the validator relayer running, topup the eth account ${txExecutor.address} on CENNZnets ${process.env.NETWORK} chain`
-                            }),
-                            httpHeader: [
-                                'Content-Type: application/json',
-                                'Accept: application/json'
-                            ],
-                        });
-                        logger.info(`Slack notification sent ${data} and status code ${statusCode}`);
-                    }
-                    return resolve();
-                }
-            });
-        } catch (e) {
-            logger.error(e);
-            return reject(e);
-        }
-    })
+// Get the notary key from CENNZnet and convert it to public key to be used to set validator on bridge contract
+async function extractNewValidators(api) {
+    const notaryKeys = await api.query.ethBridge.notaryKeys();
+    const filteredNotaryKeys = notaryKeys.filter(notaryKey => notaryKey.toString() !== IGNORE_KEY);
```

Comment:
instead of filtering `0` public key, this should return a `0` address early

Reply:
ok